### PR TITLE
Vehicle enhancements

### PR DESCRIPTION
**Pull request by Coolthulhu**
- Added z-level coord to vehicles and turret targets. Also vehicular equivalents of pos3()
- Fixed turrets failing to target monsters. Not sure how - functions are pretty much the same except 3D and it works here and not in master.
- Moved "get ammo AoE" function to damage.h/cpp, because it fits alongside `ammo_effects` (could also be useful for NPCs, not just vehicles). Also cleaned up `ammo_effects` and added a 3point argument.
- Generalized reactors to allow mods to add non-plutonium ones. Plutonium ones will function just as they did before, non-plutonium ones will consume some resource to produce some electricity.
- Removed the use of music CD in vehicle. They only ever caused trouble (CD-eating stereos, teleporting CDs when used with remote) and stereos don't really offer anything above an mp3 player.

### CODE REVIEW

**Comment by stk2008** [comment]
Nice PR dude loving your work :)

File: src/vehicle.cpp [hunk 1]
```diff
@@ -5624,7 +5551,7 @@ bool vehicle::automatic_fire_turret( int p, const itype &gun, const itype &ammo,
     // Drain a ton of power
     tmp_ups.charges = drain( fuel_type_battery, 1000 );
     tmp.worn.insert( tmp.worn.end(), tmp_ups );
-    tmp.fire_gun( xtarg, ytarg, burst );
+    tmp.fire_gun( targ, abs( parts[p].mode ) );
```
Comment (BevapDin) [inline src/vehicle.cpp]: Is the behavior change intended? The original code had `abs(mode)>1`, this here is `abs(mode)>0`.

Comment (Coolthulhu) [inline src/vehicle.cpp]: Oops, a typo

File: src/savegame_json.cpp [hunk 2]
```diff
@@ -1455,7 +1459,6 @@ void vehicle::deserialize(JsonIn &jsin)
     data.read("turn_dir", turn_dir);
     data.read("velocity", velocity);
     data.read("cruise_velocity", cruise_velocity);
-    data.read("music_id", music_id);
```
Comment (BevapDin) [inline src/savegame_json.cpp]: The vehicle class still contains this member, but it's now unused, why not remove it?

**Comment by Rivet-the-Zombie** [comment]
>Removed the use of music CD in vehicle. They only ever caused trouble (CD-eating stereos, teleporting CDs when used with remote) and stereos don't really offer anything above an mp3 player.

Striking a blow for common sense!

**Comment by SeanMirrsen** [comment]
I /liked/ the music CDs. :\ Nice immersion detail.
Make the stereo require a USB stick with music to be plugged in then. Without returning, just make it load like ammunition.

**Comment by DavidKeaton** [comment]
I liked the CDs myself. I imagine thumping over bodies listening to some 80s hair metal.

**Comment by Coolthulhu** [comment]
I removed them because they are an ugly placeholder with a bad implementation. If someone wrote a proper one that doesn't eat CDs, can keep the CD while not playing and works nicely in vehicles with stereo started in some other way than player using it (mapgen or remote control), it could be a nice addition.
Current CDs are basically just inventory clutter. Each CD is the same, they're way rarer than IRL, they limit the car stereos which are already inferior to mp3 players, they have to be ejected every time you stop playing music, you have to carry them on your person etc.

**Comment by DavidKeaton** [comment]
I got an idea for CDs then. :-)

**Pull request merged**